### PR TITLE
Pick WSGI server to run the API

### PR DESCRIPTION
**Pull request by sr-gi**
`gunicorn` is used as the WSGI server to run the api by default, however, in other setups it may make sense using something different:

- Windows machines cannot run gunicorn, so `uWSGI` may be an alternative.
- For testing and for setups where only a few (trusted) users are using the tower, Flask may suffice.

This PR adds the config parameters needed to allow the customisation of the WSGI server and adapt the daemon to run it.

partially covers #191  (not covers `uWSGI` for Windows environments, will leave as a followup since I cannot tests it personally).

This also fixes the issue with `e2e` testing leaving the `gunicorn` running. End-to-end test switched back to use `flask` now there's an option for it.

### CODE REVIEW

File: teos/api.py [hunk 1]
```diff
@@ -56,26 +56,30 @@ def get_request_data_json(request):
...
+    or by the Flask development server, in which case it is not.
 
     Args:
-        internal_api_endpoint (:obj:`str`): Endpoint where the internal api is running (host:port).
-        min_to_self_delay (:obj:`str`): The minimum to_self_delay accepted by the Inspector.
-        log_file (:obj:`str`): The file_path where to store logs.
+        internal_api_endpoint (:obj:`str`): endpoint where the internal api is running (host:port).
+        endpoint (:obj:`str`): endpoint where the http api will be running (host:port).
+        min_to_self_delay (:obj:`str`): the minimum to_self_delay accepted by the Inspector.
+        log_file (:obj:`str`): the file_path where to store logs (only necessary if not running with Flask).
```
Comment (bigspider) [inline teos/api.py:70]: What happens if not running with flask, but `log_file` is `None`? Worth adding a check?

File: teos/teosd.py [hunk 2]
```diff
@@ -227,28 +235,28 @@ def main(config):
...
+            "rpcport=",
+            "btcnetwork=",
+            "btcrpcuser=",
+            "btcrpcpassword=",
+            "btcrpcconnect=",
+            "btcrpcport=",
+            "btcfeedconnect=",
+            "btcfeedport=",
+            "datadir=",
+            "wsgi=" "daemon",
```
Comment (bigspider) [inline teos/teosd.py:254]: missing comma

File: teos/teosd.py [hunk 3]
```diff
@@ -198,16 +199,23 @@ def main(config):
...
+                    [
+                        "gunicorn",
+                        f"--bind={api_endpoint}",
+                        f"teos.api:serve(internal_api_endpoint='{INTERNAL_API_ENDPOINT}', "
+                        f"endpoint='{api_endpoint}', min_to_self_delay='{config.get('MIN_TO_SELF_DELAY')}', "
+                        f"log_file='{config.get('LOG_FILE')}')",
+                    ]
+                )
+            else:
+                os.environ["WERKZEUG_RUN_MAIN"] = "true"
```
Comment (bigspider) [inline teos/teosd.py:216]: Is it necessary to suppress the warning? After all, it's giving good advice.

**Review by bigspider — CHANGES_REQUESTED** [review]
No major comment, apart from a typo that breaks the --wsgi argument.

File: teos/api.py (hunk 1 above)
Comment (sr-gi) [reply teos/api.py]: I've changed the way this works in e7f8a437fdd766ef8744e92e425cb7df3622d60b and now the log file is required.

File: teos/teosd.py (hunk 2 above)
Comment (sr-gi) [reply teos/teosd.py]: Fixed in 54e7f24e79d8e9497fe543cfea839b6c7677453e

File: teos/teosd.py (hunk 3 above)
Comment (sr-gi) [reply teos/teosd.py]: Fixed in 54e7f24e79d8e9497fe543cfea839b6c7677453e

**Comment by sr-gi** [comment]
This should be working now. Happened to find an issue regarding python 3.8 and MacOS that extended to Windows also. It should work with versions prior to 3.8 and 3.8 onwards with the current version.

**Review by bigspider — APPROVED** [review]

**Pull request merged**